src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Main {
+
+    public static void main(String[] args) {
+
+        Figure[] figures = new Figure[6];
+        for (int i = 0; i < figures.length; i++) {
+            if (i < figures.length / 2) {
+                figures[i] = new FigureSupplier().getRandomFigure();
```

Comment:
You don't need create FigureSupplier each time. Move it to the variables before loop and use in loop